app/containers/MessageBox/index.js

```diff
@@ -187,14 +197,27 @@ class MessageBox extends Component {
 		this.setShowSend(!isTextEmpty);
 		this.handleTyping(!isTextEmpty);
 		this.setInput(text);
+		const slashCommand = text.match(/^\/([a-z0-9._-]+) (.+)/im);
+		if (slashCommand) {
+			const [, name, params] = slashCommand;
+			const command = database.objects('slashCommand').filtered('command == $0', name);
+			if (command && command[0] && command[0].providesPreview) {
+				return this.setCommandPreview(name, params);
+			}
+		}
 
 		if (!isTextEmpty) {
 			const { start, end } = this.component._lastNativeSelection;
 			const cursor = Math.max(start, end);
 			const lastNativeText = this.component._lastNativeText;
-			const regexp = /(#|@|:)([a-z0-9._-]+)$/im;
+			const regexp = /(#|@|:|^\/)([a-z0-9._-]+)$/im;
```

Comment:
We still know what this regex does, but this knowledge may be forgotten as the project grows.

Please add a comment this entire `onChangeText` flow.
It'll help in future refactors.

I won't ask you to do in this PR, but we need to fire autocomplete sooner on the other actions.
Slash commands are firing right after `/` is typed and the other actions should do the same.
Again, you don't have to do it in this PR, but keep in mind: if this two behaviours are messing up your code, consider doing so here.